File: examples/test-api.py
"""
Example Python client for GeoGrid Server
"""

import requests
import json

BASE_URL = 'http://localhost:3000'

def test_geogrid_api():
    print('🧪 Testing GeoGrid Server API')
    print('================================\n')

    try:
        # Test 1: Root endpoint
        print('📍 Test 1: Root endpoint')
        response = requests.get(f'{BASE_URL}/')
        print(json.dumps(response.json(), indent=2))
        print()

        # Test 2: Health check
        print('📍 Test 2: Health check')
        response = requests.get(f'{BASE_URL}/health')
        print(json.dumps(response.json(), indent=2))
        print()

        # Test 3: Preview presets
        print('📍 Test 3: Preview presets')
        response = requests.get(f'{BASE_URL}/api/preview/presets')
        print(json.dumps(response.json(), indent=2))
        print()

        # Test 4: Render image
        print('📍 Test 4: Render image (saving to output.png)')
        with open('examples/request-example.json', 'r') as f:
            request_data = json.load(f)

        response = requests.post(f'{BASE_URL}/api/render', json=request_data)

        with open('output.png', 'wb') as f:
            f.write(response.content)

        print(f'Status: {response.status_code}')
        print(f'Render Time: {response.headers.get("X-Render-Time")}')
        print(f'GeoRank: {response.headers.get("X-GeoRank")}')
        print(f'Grid Points: {response.headers.get("X-Grid-Points")}')
        print(f'Image size: {len(response.content)} bytes')
        print('✅ Image saved to output.png')
        print()

        # Test 5: Render base64
        print('📍 Test 5: Render base64')
        response = requests.post(f'{BASE_URL}/api/render/base64', json=request_data)
        data = response.json()

        print(f'Success: {data["success"]}')
        print(f'Metadata: {json.dumps(data["metadata"], indent=2)}')
        print(f'Metrics: {json.dumps(data["metrics"], indent=2)}')
        print(f'Base64 length: {len(data["data"])} characters')
        print()

        print('✅ All tests completed!')

    except Exception as e:
        print(f'❌ Error: {str(e)}')
        if hasattr(e, 'response'):
            print(f'Response: {e.response.text}')

if __name__ == '__main__':
    test_geogrid_api()

File: examples/test-api.js
/**
 * Example Node.js client for GeoGrid Server
 */

const axios = require('axios');
const fs = require('fs');

const BASE_URL = 'http://localhost:3000';

async function testGeoGridAPI() {
  console.log('🧪 Testing GeoGrid Server API');
  console.log('================================\n');

  try {
    // Test 1: Root endpoint
    console.log('📍 Test 1: Root endpoint');
    const rootResponse = await axios.get(`${BASE_URL}/`);
    console.log(JSON.stringify(rootResponse.data, null, 2));
    console.log('');

    // Test 2: Health check
    console.log('📍 Test 2: Health check');
    const healthResponse = await axios.get(`${BASE_URL}/health`);
    console.log(JSON.stringify(healthResponse.data, null, 2));
    console.log('');

    // Test 3: Preview presets
    console.log('📍 Test 3: Preview presets');
    const presetsResponse = await axios.get(`${BASE_URL}/api/preview/presets`);
    console.log(JSON.stringify(presetsResponse.data, null, 2));
    console.log('');

    // Test 4: Render image
    console.log('📍 Test 4: Render image (saving to output.png)');
    const requestData = require('./request-example.json');

    const imageResponse = await axios.post(`${BASE_URL}/api/render`, requestData, {
      responseType: 'arraybuffer'
    });

    fs.writeFileSync('output.png', imageResponse.data);

    console.log('Status:', imageResponse.status);
    console.log('Render Time:', imageResponse.headers['x-render-time']);
    console.log('GeoRank:', imageResponse.headers['x-georank']);
    console.log('Grid Points:', imageResponse.headers['x-grid-points']);
    console.log('Image size:', imageResponse.data.length, 'bytes');
    console.log('✅ Image saved to output.png');
    console.log('');

    // Test 5: Render base64
    console.log('📍 Test 5: Render base64');
    const base64Response = await axios.post(`${BASE_URL}/api/render/base64`, requestData);

    console.log('Success:', base64Response.data.success);
    console.log('Metadata:', base64Response.data.metadata);
    console.log('Metrics:', base64Response.data.metrics);
    console.log('Base64 length:', base64Response.data.data.length, 'characters');
    console.log('');

    console.log('✅ All tests completed!');

  } catch (error) {
    console.error('❌ Error:', error.message);
    if (error.response) {
      console.error('Response:', error.response.data);
    }
  }
}

// Run tests
testGeoGridAPI();
